### ISSUE TITLE
ChooseSimDialog: incorrect way of getting phone number

### ISSUE THREAD

**Issue by devansh-299** [openMF/mobile-wallet#892]
**Summary:**

Currently, in `ChooseSimDialog` we are selecting the sim cards and plan to use the selected sim card to fetch the phone number for phone number verification. This approach may fail for certain mobile devices and certain network providers (as explained [here](https://stackoverflow.com/questions/5134398/telephonymanager-getline1number-failing/5134725#5134725)). Hence we should manually ask the user to enter the phone number instead of relying on the sim card. This approach has also been followed in similar apps like GooglePay, MiPay.

**Steps to reproduce:**

* Go to Finance -> Tap on Add Account chip
* Tap on any bank
* `ChooseSimDialog` appears

**Comment by devansh-299** [comment]
@luckyman20 @shiv07tiwari what are your thoughts

**Comment by luckyman20** [comment]
@devansh-299 Yes, I agree on this one. There might be some inconsistencies which can create a possible blocker for the user to use our app.

**Comment by devansh-299** [comment]
> 
> 
> @devansh-299 Yes, I agree on this one. There might be some inconsistencies which can create a possible blocker for the user to use our app.

Sir, can I work on this issue?

**Comment by luckyman20** [comment]
Please first describe your approach in length.

**Comment by devansh-299** [comment]
> 
> 
> Please first describe your approach in length.

@luckyman20 , instead of relying on selecting the sim, and trying to fetch the phone numbe from the sim, we can directly use the number stored in the `PreferenceHelper`. If the user wants to change to mobile number (similar to selecting different sim card on the device), we can directly ask the user to enter the mobile number in EditText. This approach has also been followed in apps like GPay and MifPay

**Comment by devansh-299** [comment]
> 
> 
> > Please first describe your approach in length.
> 
> @luckyman20 , instead of relying on selecting the sim, and trying to fetch the phone numbe from the sim, we can directly use the number stored in the `PreferenceHelper`. If the user wants to change to mobile number (similar to selecting different sim card on the device), we can directly ask the user to enter the mobile number in EditText. This approach has also been followed in apps like GPay and MifPay

@luckyman20 should I proceed with this approach?

**Comment by luckyman20** [comment]
@shiv07tiwari @naman14 What are your thoughts?

**Comment by divyank00** [comment]
Can I work on this issue @luckyman20 ? I also believe asking user for mobile number would be a better method rather than relying on sim settings

**Comment by therajanmaurya** [comment]
Fixed in compose implementation
